File: 118-pascals-triangle/pascals-triangle.py
class Solution(object):
    def generate(self, numRows):
        """
        :type numRows: int
        :rtype: List[List[int]]
        """

        pascal = []

        for row in range(numRows):
            line = []
            res = 1
            for coloumn in range(row+1):
                if (coloumn == 0 or coloumn == row):
                    line.append(1)
                else:
                    res = res*((row-coloumn)+1)
                    res = res//coloumn
                    line.append(res)
            pascal.append(line)
        return pascal
        

File: 118-pascals-triangle/pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        int res = 1;
        vector<vector<int>>ans;
        
        for (int row = 0; row < numRows; row++){
            vector<int>level;
            for (int col = 0; col <= row; col++){
                if (col == 0 || col == row){
                    level.push_back(1);
                }
                else{
                    res = res * (row-col+1);
                    cout<<res<<endl;
                    res = res/col;
                    level.push_back(res);
                    
                }
            }
            ans.push_back(level);
            res = 1;
        }

        return ans;
    }
};